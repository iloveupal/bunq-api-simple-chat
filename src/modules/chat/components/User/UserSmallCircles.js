import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import styled from 'styled-components';

import UserColorCircle from 'Modules/chat/components/User/UserColorCircle';

import { getUsers } from 'Modules/chat/domains/users/UsersSelectors';


const CirclesContainer = styled.div`
    display: flex;
    flex-flow: row nowrap;
    padding-top: 5px;
`;

const ItemContainer = styled.div`
    padding-right: 5px;
`;

const mapStateToProps = (state) => ({
    usersData: getUsers(state),
});


export class UserSmallCircles extends PureComponent {
    static propTypes = {
        users: PropTypes.array.isRequired,
        usersData: PropTypes.object.isRequired,
    }

    render () {
        return (
            <CirclesContainer>
                {
                    this.props.users.map(( { userid } ) => (
                        <ItemContainer key={userid}>
                            <UserColorCircle
                                id={userid}
                                name={this.props.usersData[userid].name}
                                small
                            />
                        </ItemContainer>
                    ))
                }
            </CirclesContainer>
        )
    }
}

export default connect(mapStateToProps)(UserSmallCircles);