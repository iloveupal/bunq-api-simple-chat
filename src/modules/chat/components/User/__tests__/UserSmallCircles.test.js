import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { UserSmallCircles } from '../UserSmallCircles';


describe('UserSmallCircles', () => {
    it('should match snapshot', () => {
        const wrapper = mount(
            <UserSmallCircles
                users={[{ userid: '1' }, { userid: '2' }]}
                usersData={{
                    '1': {
                        name: 'Test',
                    },
                    '2': {
                        test: 'Test 2',
                    }
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});