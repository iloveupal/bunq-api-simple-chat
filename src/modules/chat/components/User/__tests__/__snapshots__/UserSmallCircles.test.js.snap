// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSmallCircles should match snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-color: #eeeeee;
  font-weight: bold;
  font-family: sans-serif;
  font-size: 10px;
  color: white;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  padding-top: 5px;
}

.c1 {
  padding-right: 5px;
}

<UserSmallCircles
  users={
    Array [
      Object {
        "userid": "1",
      },
      Object {
        "userid": "2",
      },
    ]
  }
  usersData={
    Object {
      "1": Object {
        "name": "Test",
      },
      "2": Object {
        "test": "Test 2",
      },
    }
  }
>
  <UserSmallCircles__CirclesContainer>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "UserSmallCircles__CirclesContainer-m5vu5z-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;flex-flow:row nowrap;padding-top:5px;",
            ],
          },
          "displayName": "UserSmallCircles__CirclesContainer",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "UserSmallCircles__CirclesContainer-m5vu5z-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <UserSmallCircles__ItemContainer
          key="1"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "UserSmallCircles__ItemContainer-m5vu5z-1",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "padding-right:5px;",
                  ],
                },
                "displayName": "UserSmallCircles__ItemContainer",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "UserSmallCircles__ItemContainer-m5vu5z-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <UserColorCircle
                id="1"
                name="Test"
                small={true}
              >
                <UserColorCircle__ColorCircle
                  color="#eeeeee"
                  small={true}
                  title="Test"
                >
                  <StyledComponent
                    color="#eeeeee"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "display:flex;flex-flow:row nowrap;justify-content:center;align-items:center;width:",
                            [Function],
                            ";height:",
                            [Function],
                            ";border-radius:",
                            [Function],
                            ";background-color:",
                            [Function],
                            ";font-weight:bold;font-family:sans-serif;font-size:",
                            [Function],
                            ";color:white;cursor:pointer;",
                          ],
                        },
                        "displayName": "UserColorCircle__ColorCircle",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    small={true}
                    title="Test"
                  >
                    <div
                      className="c2"
                      color="#eeeeee"
                      title="Test"
                    >
                      T
                    </div>
                  </StyledComponent>
                </UserColorCircle__ColorCircle>
              </UserColorCircle>
            </div>
          </StyledComponent>
        </UserSmallCircles__ItemContainer>
        <UserSmallCircles__ItemContainer
          key="2"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "UserSmallCircles__ItemContainer-m5vu5z-1",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "padding-right:5px;",
                  ],
                },
                "displayName": "UserSmallCircles__ItemContainer",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "UserSmallCircles__ItemContainer-m5vu5z-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <UserColorCircle
                id="2"
                small={true}
              >
                <UserColorCircle__ColorCircle
                  color="#eeeeee"
                  small={true}
                >
                  <StyledComponent
                    color="#eeeeee"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "display:flex;flex-flow:row nowrap;justify-content:center;align-items:center;width:",
                            [Function],
                            ";height:",
                            [Function],
                            ";border-radius:",
                            [Function],
                            ";background-color:",
                            [Function],
                            ";font-weight:bold;font-family:sans-serif;font-size:",
                            [Function],
                            ";color:white;cursor:pointer;",
                          ],
                        },
                        "displayName": "UserColorCircle__ColorCircle",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    small={true}
                  >
                    <div
                      className="c2"
                      color="#eeeeee"
                    />
                  </StyledComponent>
                </UserColorCircle__ColorCircle>
              </UserColorCircle>
            </div>
          </StyledComponent>
        </UserSmallCircles__ItemContainer>
      </div>
    </StyledComponent>
  </UserSmallCircles__CirclesContainer>
</UserSmallCircles>
`;
